eth/fetcher/fetcher.go

```diff
@@ -338,9 +338,15 @@ func (f *Fetcher) loop() {
 			if _, ok := f.fetching[notification.hash]; ok {
 				break
 			}
+			if _, ok := f.fetched[notification.hash]; ok {
+				break
+			}
 			if _, ok := f.completing[notification.hash]; ok {
 				break
 			}
+			if _, ok := f.queued[notification.hash]; ok {
+				break
+			}
```

Comment:
Similar to the above, `completing` will contain the block up to the point of that being successfully imported into the chain, so there's no need to check the contents of `queued` too here.